lib/Reader.php

```diff
@@ -169,7 +171,8 @@ public function parseInnerTree(array $elementMap = null)
                     case self::END_ELEMENT:
                         // Ensuring we are moving the cursor after the end element.
                         $this->read();
-                        break 2;
```

Comment:
This `break 2` was causing phpstan to report:
```
Line 187    Unreachable statement - code above always terminates.
```
phpstan seems to not realise that `break 2` is a way out of both the `switch` and `while` statements.

It is easy enough to refactor the code to have a boolean to control the `while` loop. And maybe that makes it easier to read?

Reply:
we should report a upstream issue.. in case you dont want/don't have time to do it, i will do it.

just let me know.

Reply:
@staabm you know better exactly about upstream. Please report.

Reply:
will do. thx.

IMO the newly introduced logic is better to read, because its now obvious what this single spot is trying to achieve. the `break 2` could also be read as a mistake

Reply:
lets see: https://github.com/phpstan/phpstan/issues/2926